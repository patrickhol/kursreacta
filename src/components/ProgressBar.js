import React from 'react'
export const ProgressBar = ({ className, percent }) => {
    return (<div className={`progress-bar ${className}`}>
        <div style={{ width: `${percent}%` }}></div>
    </div>);
}
