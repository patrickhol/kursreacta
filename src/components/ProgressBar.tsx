import React from 'react';
import classN from 'classnames';

interface ProgressBarInterface {
  className: string;
  percent: number;
}

export const ProgressBar = ({className, percent}: ProgressBarInterface) => {
  const progressClassName = classN('progress-bar', className);

  return (
    <div className={progressClassName}>
      <div style={{width: `${percent}%`}}></div>
    </div>
  );
};
