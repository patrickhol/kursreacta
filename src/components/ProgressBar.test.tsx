import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import {ProgressBar} from './ProgressBar';

var clockRenderer: any = null;
var root: any = null;

describe('<ProgressBar>', () => {
  describe('Given classname and percent', () => {
    beforeEach(() => {
      root = document.createElement('div');
      ReactDOM.render(
        <ProgressBar className={'inactive'} percent={90} />,
        root,
      );
    });
    it('then renders div', () => {
      expect(root.childNodes[0].nodeName).toEqual('DIV');
    });
    it('then renders classNames cprogress-bar inactive/ ', () => {
      expect(root.childNodes[0].className).toMatch(/progress-bar inactive/);
    });
  });

  describe('Given percent', () => {
    beforeEach(() => {
      root = document.createElement('div');
      ReactDOM.render(<ProgressBar className={''} percent={90} />, root);
    });
    it('then renders H2', () => {
      expect(root.childNodes[0].nodeName).toEqual('DIV');
    });
    it('then renders classNames clock inactive ', () => {
      expect(root.childNodes[0].className).toMatch(/progress-bar/);
    });
  });

  describe('Given className and percent (TestRenderer)', () => {
    beforeEach(() => {
      clockRenderer = renderer.create(
        <ProgressBar className={'inactive'} percent={90} />,
      );
    });
    it('then renders div', () => {
      expect(clockRenderer.toJSON().type).toEqual('div');
    });
    it('then renders classNames progress-bar inactive', () => {
      expect(clockRenderer.toJSON().props).toMatchObject({
        className: expect.stringMatching(/progress-bar inactive/),
      });
    });

    it('then renders snapshot Clock ', () => {
      expect(clockRenderer.toJSON()).toMatchSnapshot();
    });
  });
});
