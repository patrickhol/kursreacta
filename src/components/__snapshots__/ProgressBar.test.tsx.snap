// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar> Given className and percent (TestRenderer) then renders snapshot Clock  1`] = `
<div
  className="progress-bar inactive"
>
  <div
    style={
      Object {
        "width": "90%",
      }
    }
  />
</div>
`;
